src/Symfony/Component/Lock/Store/MysqlStore.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MysqlStore is a StoreInterface implementation using MySQL/MariaDB GET_LOCK function.
+ *
+ * @author Jérôme TAMARELLE <jerome@tamarelle.net>
+ */
+class MysqlStore implements StoreInterface
+{
+    private $dsn;
+    private $username;
+    private $password;
+    private $options;
+    private $waitTimeout;
+
+    /**
+     * List of available options:
+     *  * db_username: The username when lazy-connect [default: '']
+     *  * db_password: The password when lazy-connect [default: '']
+     *  * db_connection_options: An array of driver-specific connection options [default: array()]
+     *  * wait_timeout: Time in seconds to wait for a lock to be released. A negative value means infinite. [default: -1].
+     *
+     * @param string $dsn     The connection DSN string
+     * @param array  $options configuration options
+     */
+    public function __construct($dsn, array $options)
+    {
+        $this->dsn = $dsn;
+        $this->username = $options['db_username'] ?? '';
+        $this->password = $options['db_password'] ?? '';
+        $this->options = $options['db_connection_options'] ?? array();
+        $this->waitTimeout = $options['wait_timeout'] ?? -1;
```

Comment:
In others stores blocking mode is waiting forever. Such option is not needed IMO

Reply:
The "wait timeout" feature was not available in flock and semaphore, isn't it the reason why it was not implemented? It is a useful alternative to `RetryTillSaveStore` when the lock needs to be acquired in a limited time with a clean error handling.

Reply:
It wasn't implemeted in the store because I didn't see end to end use case, and such feature can be handled by a decorator and bring the feature to every store.